File: Reverse a String - GFG/reverse-a-string.java
// { Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.io.*;
import java.lang.*;

class Driver
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        
        while(t-- >0)
        {
            String str = read.readLine();
            System.out.println(new Reverse().reverseWord(str));
        }
    }
}// } Driver Code Ends


//User function Template for Java


class Reverse
{
    // Complete the function
    // str: input string
    public static String reverseWord(String str)
    {
        char[] data = str.toCharArray();    
        
        for(int i=0;i<data.length/2; i++){
            char temp = data[i];
            data[i] = data[data.length-1-i];
            data[data.length-1-i] = temp;
        }
        
        return String.valueOf(data);
    }
}

File: Reverse a String - GFG/reverse-a-string.cpp
// { Driver Code Starts
//Initial Template for C++


#include<bits/stdc++.h>
using namespace std;


string reverseWord(string str);


int main() {
	
	int t;
	cin>>t;
	while(t--)
	{
	string s;
	cin >> s;
	
	cout << reverseWord(s) << endl;
	}
	return 0;
	
}

// } Driver Code Ends


//User function Template for C++

string reverseWord(string str){
    int n = str.length();
    int low = 0;
    int high = n-1;
    while(low<high){
        swap(str[low],str[high]);
        low++;
        high--;
    }
    
    return str;
}